datajunction-server/datajunction_server/models/access.py

```diff
@@ -0,0 +1,436 @@
+                        message="Injected `validate_access` must approve or deny all requests.",
+                    ),
+                ],
+            )
+
+        return self.validation_results
+
+    def validate_and_raise(self):
+        """
+        Validates with ACS and raises if any resources were denied
+        """
+        self.validate()
+        self.raise_if_invalid_requests()
+
+
+def validate_access_nodes(
+    validate_access: "ValidateAccessFn",  # pylint: disable=W0621
+    verb: ResourceRequestVerb,
+    user: Optional[User],
+    nodes: Iterable[Node | NodeRevision],
+    raise_: bool = False,
+) -> List[NodeRevision | Node]:
+    """
+    Validate the access of the user to a set of nodes
+    """
+    if user is None:
+        raise DJException(#pragma: no cover
+            http_status_code=HTTPStatus.FORBIDDEN,
+            errors=[
+                DJError(
+                    code=ErrorCode.INCOMPLETE_AUTHORIZATION,
+                    message="Failed to acquire user to validate access.",
+                ),
+            ],
+        )
+    access_control = AccessControlStore(
+        validate_access=validate_access,
+        user=user,
+    )
+
+    access_control.add_request_by_nodes(verb, nodes)
+
+    validation_results = access_control.validate()
+    if raise_:
+        access_control.raise_if_invalid_requests()#pragma: no cover
+
+    return [
+        node
+        for node in nodes
+        if node.id
+        in {
+            request.access_object.revision_id
+            for request in validation_results
+            if request.approved
+        }
+    ]
+
+
+def validate_access_namespaces(
+    validate_access: "ValidateAccessFn",  # pylint: disable=W0621
+    verb: ResourceRequestVerb,
+    user: Optional[User],
+    namespaces: Iterable[str],
+    raise_: bool = False,
+) -> List[str]:
+    """
+    Validate the access of the user to a set of namespaces
+    """
+    if user is None:
+        raise DJException(#pragma: no cover
+            http_status_code=HTTPStatus.FORBIDDEN,
+            errors=[
+                DJError(
+                    code=ErrorCode.INCOMPLETE_AUTHORIZATION,
+                    message="Failed to acquire user to validate access.",
+                ),
+            ],
+        )
+    access_control = AccessControlStore(
+        validate_access=validate_access,
+        user=user,
+    )
+    for namespace in namespaces:
+        access_control.add_request_by_namespace(verb, namespace)
+
+    validation_results = access_control.validate()
+    if raise_:
+        access_control.raise_if_invalid_requests()#pragma: no cover
+
+    return [
+        namespace
+        for namespace in namespaces
+        if namespace
+        in {
+            request.access_object.name
+            for request in validation_results
+            if request.approved
+        }
+    ]
+
+
+ValidateAccessFn = Callable[[AccessControl], None]
+
+
+# Dummy default if not dependency injected
```

Comment:
This comment can just go in the docstring

Reply:
I was afraid it might carry onto someone implementing their own and confuse them

Reply:
Ah I see, that's a good point. Maybe it can be reworded to get the same message across but also provide clarity? Such as describing it as a placeholder/example access validation dependency that approves all requests.